### PR TITLE
Don't rely on new lines in IPC RPC (soltest)

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: test/RPCSession.h [hunk 1]
```diff
@@ -47,7 +46,8 @@ class IPCSocket : public boost::noncopyable
 
 private:
 	std::string m_path;
-	HANDLE  m_socket;
+	HANDLE m_socket;
+	TCHAR m_readBuf[512000];
```
Comment (axic) [inline test/RPCSession.h:50]: Moving this from the method, from being a stack variable to be on the heap. Also reduced from 5Mb to 512k.

**Comment by axic** [comment]
(This might also fix #1664)

**Review by VoR0220 — APPROVED** [review]
Subject to appveyor finishing but this gets my approval thus far.

**Comment by VoR0220** [comment]
@axic just tested and nope, does not fix it.

**Comment by axic** [comment]
> @axic just tested and nope, does not fix it.

@VoR0220 It seems that 0 bytes are returned (note the other lines have a new line because that is not cut off). That suggests `eth` crashed in the docker instance.

**Comment by VoR0220** [comment]
@axic perhaps it needs more time to boot? Or maybe it needs to be initialized with more memory in travis....not sure either way.

File: test/RPCSession.cpp [hunk 2]
```diff
@@ -116,18 +111,15 @@ string IPCSocket::sendRequest(string const& _req)
...
-	{
-		if (c != '\n')
-			response += c;
-		else
-			break;
-	}
-	return response;
+	ssize_t ret = recv(m_socket, m_readBuf, sizeof(m_readBuf), 0);
+
+	if (ret < 0)
```
Comment (axic) [inline test/RPCSession.cpp]: This should be <=0 to detect closed sockets.

**Comment by axic** [comment]
I think this can be merged once Appveyor passes again.

**Review by chriseth — APPROVED** [review]

**Pull request merged**